server/api/crud/model_monitoring/model_endpoints.py

```diff
@@ -190,13 +197,49 @@ def patch_model_endpoint(
 
         return self._convert_into_model_endpoint_object(endpoint=model_endpoint_record)
 
+    @staticmethod
+    def _get_features(
+        model: mlrun.artifacts.ModelArtifact,
+        project: str,
+        run_db: server.api.rundb.sqldb.SQLRunDB,
+    ) -> list[mlrun.feature_store.Feature]:
+        """Get features to the feature set according to the model object"""
+        features = []
+        if model.spec.inputs:
+            for feature in itertools.chain(model.spec.inputs, model.spec.outputs):
+                name = mlrun.feature_store.api.norm_column_name(feature.name)
+                features.append(
+                    mlrun.feature_store.Feature(
+                        name=name, value_type=feature.value_type
+                    )
+                )
+        # Check if features can be found within the feature vector
+        elif model.spec.feature_vector:
+            _, name, _, tag, _ = mlrun.utils.helpers.parse_artifact_uri(
+                model.spec.feature_vector
+            )
+            fv = run_db.get_feature_vector(name=name, project=project, tag=tag)
+            for feature in fv.status.features:
+                if feature["name"] != fv.status.label_column:
+                    name = mlrun.feature_store.api.norm_column_name(feature["name"])
+                    features.append(
+                        mlrun.feature_store.Feature(
+                            name=name, value_type=feature["value_type"]
+                        )
+                    )
+        else:
+            logger.warn(
+                "Could not find any features in the model object and in the Feature Vector"
+            )
+        logger.debug("Listed features", features=features)
+        return features
+
     @staticmethod
     def create_monitoring_feature_set(
+        features: list[mlrun.feature_store.Feature],
         model_endpoint: mlrun.common.schemas.ModelEndpoint,
-        model_obj: mlrun.artifacts.ModelArtifact,
         db_session: sqlalchemy.orm.Session,
-        run_db: server.api.rundb.sqldb.SQLRunDB,
-    ):
```

Comment:
update the docstring of `create_monitoring_feature_set` (remove `model_obj` and `run_db` + add `features`)

Reply:
Good catch - fixed: 19d0184